### PR TITLE
[FLINK-3697] Properly access type information for nested POJO key selection

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/util/FieldAccessor.java [hunk 1]
```diff
@@ -182,11 +182,14 @@ public R set(R record, F fieldValue) {
 			@SuppressWarnings("unchecked")
 			CompositeType<R> cType = (CompositeType<R>) type;
 
+			if(field.contains(".")) {
+				throw new IllegalArgumentException("The Pojo field accessor currently doesn't support nested POJOs");
```
Comment (RonCrocker) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/util/FieldAccessor.java]: Note that the problem reported in [FLINK-3697](https://issues.apache.org/jira/browse/FLINK-3697) was not related to accessing a nested field.. Perhaps there should be a subsequent JIRA ticket to support nested POJO accessors.

Comment (rmetzger) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/util/FieldAccessor.java]: Thank you for reviewing my pull request. I've opened FLINK-3702 for that.

See also here: https://github.com/apache/flink/pull/1851/files#diff-e5e091a2c1e4bf850ef93e2010fe4c81R132

**Comment by RonCrocker** [comment]
:+1:

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamPojoITCase.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	/**
+	 * Test composite key on the Data POJO (with nested fields)
+	 */
+	@Test
+	public void testCompositeKeyOnNestedPojo() throws Exception {
+		StreamExecutionEnvironment see = StreamExecutionEnvironment.getExecutionEnvironment();
+		see.getConfig().disableObjectReuse();
+		see.setParallelism(3);
+
+		DataStream<Data> dataStream = see.addSource(new DataGenerator());
```
Comment (StephanEwen) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamPojoITCase.java]: Why not use "fromCollection()" here and have a prepared list of sample data...

**Comment by rmetzger** [comment]
I'll address Stephan's comment and merge the change

**Pull request closed without merging**